src/createPackage/models/createPackageManager.ts

```diff
@@ -103,118 +77,68 @@ export class CreatePackageManager {
-    for (let i = 0; i <= zoomLevel; i++) {
-      batches.push(bboxToTileRange(sanitizedBbox as BBox2d, i));
-    }
-    // const batches = this.generateTileGroups(polygon as Polygon, layerMetadata.footprint as Polygon | MultiPolygon, zoomLevel);
-    const estimatesGpkgSize = calculateEstimateGpkgSize(batches, tileEstimatedSize); // size of requested gpkg export
-    if (this.storageEstimation.validateStorageSize) {
-      const isEnoughStorage = await this.validateFreeSpace(estimatesGpkgSize);
-      if (!isEnoughStorage) {
-        throw new InsufficientStorage(`There isn't enough free disk space to executing export`);
+    const processingExists = await this.checkForExportProcessing(dupParams, callbackUrls);
+    if (processingExists) {
+      // For race condition
+      completedExists = await this.checkForExportCompleted(dupParams);
+      if (completedExists) {
+        return completedExists;
       }
+      return { ...processingExists, isDuplicated: true };
     }
-    const separator = this.getSeparator();
-    const packageName = this.generatePackageName(productType, resourceId, version, zoomLevel, sanitizedBbox);
-    const packageRelativePath = getGpkgRelativePath(packageName, separator);
-    const sources: IMapSource[] = [
-      {
-        path: packageRelativePath,
-        type: 'GPKG',
-        extent: {
-          minX: sanitizedBbox[0],
-          minY: sanitizedBbox[1],
-          maxX: sanitizedBbox[2],
-          maxY: sanitizedBbox[3],
-        },
-      },
-      {
-        path: `${layerMetadata.id as string}${separator}${layerMetadata.displayPath as string}`, //tiles path
-        type: this.tilesProvider,
-      },
-    ];
 
-    const workerInput: IWorkerInput = {
-      sanitizedBbox,
-      targetResolution,
-      fileName: packageName,
-      relativeDirectoryPath: getGpkgNameWithoutExt(packageName),
-      zoomLevel,
-      dbId,
-      exportVersion: ExportVersion.GETMAP,
-      version: version,
-      cswProductId: resourceId,
-      crs: crs ?? DEFAULT_CRS,
-      productType,
-      batches,
-      sources,
-      priority: priority ?? DEFAULT_PRIORITY,
-      callbacks: callbacks,
-      gpkgEstimatedSize: estimatesGpkgSize,
-      targetFormat: layerMetadata.tileOutputFormat,
-      outputFormatStrategy: TileFormatStrategy.MIXED,
-    };
-    const jobCreated = await this.jobManagerClient.create(workerInput);
-    return jobCreated;
+    return undefined;
   }
 
   public async createPackageRoi(userInput: ICreatePackageRoi): Promise<ICreateExportJobResponse | ICallbackExportResponse> {
     const { dbId, crs, priority, callbackURLs, description } = userInput;
     let roi = userInput.roi;
     const layer = await this.rasterCatalogManager.findLayer(userInput.dbId);
     const layerMetadata = layer.metadata;
+
+    const spanOptions: SpanOptions = {
+      kind: SpanKind.PRODUCER,
+    };
+    const mainSpan = this.tracer.startSpan('jobManager.job create', spanOptions);
```

Comment:
![image](https://github.com/user-attachments/assets/837abc63-5a69-4f4b-a53a-6bf95cac8e80)

Reply:
missed it.